Subject: [PATCH] Fix error handling problems

There are
 1) a potential memory leak by reduceDescriptions.
 2) potential null-pointer-accesses by pragmaKeys and pragmaValues.

diff --git a/cpu_ref/rsCpuExecutable.cpp b/cpu_ref/rsCpuExecutable.cpp
index 3d5e635..8eb2065 100644
--- a/cpu_ref/rsCpuExecutable.cpp
+++ b/cpu_ref/rsCpuExecutable.cpp
@@ -697,15 +697,24 @@
 
 #ifndef RS_COMPATIBILITY_LIB
 
-    for (size_t idx = 0; idx < pragmaCount; ++idx) {
-        delete [] pragmaKeys[idx];
-        delete [] pragmaValues[idx];
+    if (pragmaKeys) {
+        for (size_t idx = 0; idx < pragmaCount; ++idx) {
+            delete [] pragmaKeys[idx];
+        }
+    }
+
+    if (pragmaValues) {
+        for (size_t idx = 0; idx < pragmaCount; ++idx) {
+            delete [] pragmaValues[idx];
+        }
     }
 
     delete[] pragmaValues;
     delete[] pragmaKeys;
 #endif  // RS_COMPATIBILITY_LIB
 
+    delete[] reduceDescriptions;
+
     delete[] forEachSignatures;
     delete[] forEachFunctions;
 
